base/loading.jl

```diff
@@ -2503,7 +2503,13 @@ end
 # load a serialized file directly from append_bundled_depot_path for uuidkey without stalechecks
 function require_stdlib(uuidkey::PkgId, ext::Union{Nothing, String}=nothing)
     @lock require_lock begin
+    if ext isa String
+        uuidkey = PkgId(uuid5(uuidkey.uuid, ext), ext)
+    end
     if root_module_exists(uuidkey)
```

Comment:
I believe you are supposed to call `startloading`, but this assumption that `__init__` finished running has never been enforced or made true anywhere else at any time, so it seems fairly sketchy to assume it is enforced here.
```suggestion
    if root_module_exists(uuidkey)
```

Reply:
Ok. if that's a bigger question, then I'll just directly handle `package_locks` in https://github.com/JuliaLang/julia/pull/54674

Reply:
Yeah, I don't think we've enforced it, but I do think that it's reasonable to start enforcing it. It would be weird if two threads start doing `using Foo`, which spins up some C state in its `__init__`, but for one of them `using` returns before that happens.